src/blockdata/transaction.rs

```diff
@@ -689,6 +707,21 @@ impl SigHashType {
          }
      }
 
+     /// Read a 4-byte uint32 as a standard sighash type, returning an error if the type
+     /// is non standard.
+     pub fn from_u32_standard(n: u32) -> Result<SigHashType, ParseSigHashTypeError> {
+         match n & 0x9f {
```

Comment:
Why do we have the `9f` mask? Should it not be `ff`? It seems that bitcoin core would call 0xa1 as non-standard, but it would make it standard

Should we also be enforcing that the three bytes are zero here?

Reply:
After discussion with Andrew on IRC it is a bug in the original code, pushing a fixing commit for it then removing from my code.

Nice catch, i should have questioned it..